### PR TITLE
Add logic for resharding

### PR DESCRIPTION
**Pull request by mikedoan**
## Issue

Add logic for resharding. I have to stop in the middle because i need to modify resharding calculation logic which currently does not calculate node role (leader or validator) but instead it only calculates new committees.

### CODE REVIEW

**Comment by LeoHChen** [comment]
still fail travis checker.

**Comment by harmony-ek** [comment]
> still fail travis checker.

Yep.  Two issues:

1. goimports ended up tidying one file; to fix, run scripts/goimports.sh on your side to format the file,  then commit/push it.
2. For some reason, gomock isn't installed on Travis, so mockgen is failing (and mock freshness check is skipped as a result).  I'll fix this.

File: node/service_setup.go [hunk 1]
```diff
@@ -42,6 +35,16 @@ func (node *Node) setupForShardLeader() {
 	node.serviceManager.RegisterService(service.Randomness, randomness.New(node.DRand))
 }
 
+func (node *Node) stopShardLeaderServices() {
```
Comment (rlan35) [inline node/service_setup.go]: These stop funcs seems pretty verbose. Is there a way to loop through the services and and stop signal iteratively?

Comment (mikedoan) [reply node/service_setup.go]: yep. this is better. was thinking about this.

File: api/service/resharding/service.go [hunk 2]
```diff
@@ -49,6 +58,18 @@ func (s *Service) Run(stopChan chan struct{}, stoppedChan chan struct{}) {
 
 // DoService does role conversion.
 func (s *Service) DoService() {
+	tick := time.NewTicker(ReshardingCheckTime)
```
Comment (chaosma) [inline api/service/resharding/service.go]: Resharding is block number based instead of time based. Do we really need a service?

Comment (mikedoan) [reply api/service/resharding/service.go]: Regarding my PR, the resharding service keeps checking the new  status from beacon blockchain - currently i designs it checking every single second.

Is there any suggestionof not having a service?

Comment (mikedoan) [reply api/service/resharding/service.go]: FYI, this is still an ongoing work

**Pull request merged**